### ISSUE TITLE
One time code for authenticaiton

### ISSUE THREAD

**Issue by hifabienne** [zitadel/zitadel#6099]
As a administrator I want to enable my users to be able to authenticate them with a one time code, so they do not need a password or preconfigured authentication method. This could also be used as account recovery, e.g when a linked account was deleted or a device got lost.

**Acceptance Criteria**
- [ ] I am able to use one time code for authentication
- [ ] I am able to configure on an instance and organization if this authentication method is possible
- [ ] I am able to set if the code should be sent by ZITADEL per email or phone, or if I want to code back in the response, so I can send it myself

**Comment by jschwertfeger** [comment]
Functionality really needed/great for soft onboarding. 👍

**Comment by hifabienne** [comment]
> Functionality really needed/great for soft onboarding. 👍

Hi @jschwertfeger 
We do not have this yet in our hosted login ui.
But if you like to build your own Login UI with our ressource APIs, you could already do that by using the email code.
We have a description here how to use it for MFA, but you could also use it to directly send the code and skip the password:
https://zitadel.com/docs/guides/integrate/login-ui/mfa#email-code-registration

**Comment by mffap** [comment]
+1 from inbound

@hifabienne would it make sense to add Magic Link in the title for users to find this feature quickly?

**Comment by mffap** [comment]
+1 from inbound

**Comment by hifabienne** [comment]
@livio-a @muhlemmer 
Can you please estimate this issue?

**Comment by livio-a** [comment]
just as a note: discussed (also with @hifabienne) if v2 session api would be sufficient

**Comment by muhlemmer** [comment]
We can estimate this now, don't we @livio-a?

**Comment by tbenyon** [comment]
Sadly I think missing Magic Links has rendered Zitadel something I can't use for now :(

My scenario ideal flow is:
- user enters e-mail and logs in via magic link
- admin can update configurable list of domains so that any user with email domain `@myCompany.com` `@secondCompany.com` can auto register themselves as a user

**Comment by b3nk3** [comment]
I have the same need as outlined by @tbenyon. 

Have you got an ETA for this?

**Comment by hifabienne** [comment]
We have not built yet the functionality in our hosted login.
However, if you would like to have the possibility of MagicLink, you can use our session API and build the login UI yourself.
API Docs: https://zitadel.com/docs/apis/resources/session_service_v2
Build your own login: https://zitadel.com/docs/guides/integrate/login-ui

**Comment by stonymahony** [comment]
@hifabienne Hi, thanks for your hints, 'm very interested in this feature and willing to implement it with my own UI and zitadel API. I follow the instructions, [checking the email code works](https://zitadel.com/docs/guides/integrate/login-ui/mfa#check-email-code), I get a session token. 
But - what do i do with it? How do I make it so that the user is ‘really logged in’, as he would normally be via OIDC? I'm a bit lost....

**Comment by muhlemmer** [comment]
Please don't use this issue as support thread. You can use Github discussions or Discord for that.

**Comment by ArnauAregall** [comment]
> We have not built yet the functionality in our hosted login. However, if you would like to have the possibility of MagicLink, you can use our session API and build the login UI yourself. API Docs: https://zitadel.com/docs/apis/resources/session_service_v2 Build your own login: https://zitadel.com/docs/guides/integrate/login-ui

Hi @hifabienne, now that the Session API Login V2 is the default in Zitadel V4, what would be your recommendations to implement this feature?

**Comment by veritymedia** [comment]
@ArnauAregall Wouldnt this be possible without even customising the login app? We can do webhook calls, so we can do this  email sending and verification on our end, then use the Zitadel API to create a new user with the email and verified status? It maybe less "easy" and "pretty" to do, but its totally reasonable.
Correct me if im wrong.

**Comment by ArnauAregall** [comment]
> @ArnauAregall Wouldnt this be possible without even customising the login app? We can do webhook calls, so we can do this  email sending and verification on our end, then use the Zitadel API to create a new user with the email and verified status? It maybe less "easy" and "pretty" to do, but its totally reasonable.
> Correct me if im wrong.

Not sure how we would do that without any customization on the login flow. Quoting [this recent thread](https://discord.com/channels/927474939156643850/1425863128482058311) on Discord right now users need to have at least one first factor of authentication, either a password, IdP or passkey.

My team and I could see a setup like having a user with a predefined temporal password as first factor of auth, **and** a 2FA of OTP via email enabled, then email/password submit flow could be automated by the login app and then prompt her for the OTP.

It's still username+password+OTP but in terms of UX it would **feel like** a magic link. Anyhow again this requires a custom flow on the login app.

**Comment by owingruters** [comment]
I agree this is a most welcome feature. Instead of everybody creating their own implementation, it would be preferrable Zitadel gets the functionality.
Reading the above messages, it seems everybody agrees on this. So it is a matter of when! Is it on your backlog, and if so, do you have an idea when it will be done?
